pages/content/amp-dev/search.html

```diff
@@ -17,121 +17,116 @@
+<template id="search_results_template" type="amp-mustache">
+  <div class="component-item">
+    {{#result.components}}
+    <div class="item">
+      <a href="{{url}}">
+        <strong class="title">{{title}}</strong>
+        <p class="copy">{{description}}</p>
+      </a>
+      {{#exampleUrl}}
+      <a class="ap-m-lnk" href="{{exampleUrl}}">
+        <span class="ap-m-lnk-text">Show example</span>
+      </a>
+      {{/exampleUrl}}
+      {{#playgroundUrl}}
+      <a class="ap-m-lnk"href="{{playgroundUrl}}">
+        <span class="ap-m-lnk-text">Show example in playground</span>
+      </a>
+      {{/playgroundUrl}}
+    </div>
+    <hr>
+    {{/result.components}}
+  </div>
 
+  <div class="items">
+    {{#result.pages}}
+    <div class="item">
+      <a href="{{url}}">
+        <strong class="title">{{title}}</strong>
+        <p class="copy">{{description}}</p>
+      </a>
+    </div>
+    {{/result.pages}}
+  </div>
+</template>
+{% endraw %}
 
 <section class="ap--container">
 
   <div class="ap-m-copy ap-m-copy-center-small ap-m-biggy-background">
     <h1>Search</h1>
-    <form action-xhr="/search/do" class="search-form" id="searchForm" method="GET" on="submit-success: AMP.setState({'query': tempQuery, 'pageNumber': 1, 'showResults': true, 'search': event.response})"
+    <form action-xhr="/search/do" class="search-form" id="searchForm" method="GET"
+          on="submit-success: AMP.setState({'showResults': true, 'search': event.response, 'totalResults': event.response.result.totalResults})"
           target="_top">
       <amp-autocomplete filter="substring"
                         min-characters="0"
-                        on="select:AMP.setState({ 'tempQuery': event.value }),searchForm.submit"
+                        on="select:searchForm.submit"
                         submit-on-enter="false"
                         src="/search/autosuggest"
       >
-        <input name="q" on="change:AMP.setState({ 'tempQuery': event.value })"
-               placeholder="What are you looking for?"
-               required>
+        <input name="q" placeholder="What are you looking for?" required>
       </amp-autocomplete>
-      <input name="page" type="hidden" value="1">
       <input name="locale" type="hidden" value="{{ doc.locale }}">
       <button type="submit">Search</button>
+
+      <p [text]="totalResults + ' results found'" class="info"></p>
+
+      <div submit-success template="search_results_template"></div>
+      <div submit-error>
+        An internal server error occurred while trying to load the search results.<br>
+        We apologize for the inconvenience caused.<br>
+        Please try again later.
+      </div>
+
     </form>
-    <div [hidden]="showResults != true" hidden>
-      <p [text]="search.result.totalResults + ' results found'" class="info"></p>
+
+    <div [hidden]="totalResults <= 10" hidden>
 
       <amp-list [height]="(100) * search.result.components.length + (80) * search.result.pages.length"
```

Comment:
use: `is-layout-container` instead of calculating the height https://amp.dev/documentation/components/amp-list/#[is-layout-container]-(optional)